sdk/core/core-lro/package.json

```diff
@@ -34,20 +32,21 @@
     "dist/",
     "dist-esm/",
     "README.md",
-    "LICENSE"
+    "LICENSE",
+    "dist-esm/src"
```

Comment:
why are we shipping source? The sourcemaps now should have the sources embedded to avoid this requirement.

Reply:
@xirzec the linter enforces it here: https://github.com/Azure/azure-sdk-for-js/blob/4a9359cfda67218d2ed2c48e6048f0a3f921da35/common/tools/eslint-plugin-azure-sdk/src/rules/ts-package-json-files-required.ts#L68
Should we get ride of this rule?

Reply:
Absolutely, if anything we should invert this requirement to *not* include it.

See: https://github.com/Azure/azure-sdk-for-js/pull/7615

Reply:
I see. I opened https://github.com/Azure/azure-sdk-for-js/pull/11411.

Reply:
@xirzec These aren't merely source files, though. They're used by any bundlers/transpilers that are module compatible. Furthermore, browser mappings rely on the ESMs being in the bundle, IIRC.

We shouldn't bundle TS sources, since they're inlined into the ESMs, but we still need to bundle the ESMs.

Reply:
Thanks @willmtemple and @xirzec for the feedback. I added `dist-esm/src` to the `files` list. I believe we do not need to ship the `test` or `samples` esms there so I am only including the `src` subdir.